routes/database.js

```diff
@@ -8,480 +8,382 @@ let my_conf = cfgsrv.get_conf();
+        'owner': session_user.uid,
+        'date': new Date().getTime(),
+        'action': 'database ' + req.params.id + ' changed from ' + req.params.old + ' to ' + req.params.new,
+        'logs': []
+    });
+    res.send({ message: 'Owner changed from ' + req.params.old + ' to ' + req.params.new });
 });
 
 
 router.get('/database', async function(req, res) {
-    if(! req.locals.logInfo.is_logged) {
-        res.status(401).send({message: 'Not authorized'});
+    if(!req.locals.logInfo.is_logged) {
+        res.status(401).send({ message: 'Not authorized' });
         return;
     }
     let session_user = null;
-    let isadmin = false;
+    let is_admin = false;
     try {
         session_user = await dbsrv.mongo_users().findOne({_id: req.locals.logInfo.id});
-        isadmin = await rolsrv.is_admin(session_user);
+        is_admin = await rolsrv.is_admin(session_user);
     } catch(e) {
         logger.error(e);
-        res.status(404).send({message: 'User session not found'});
-        res.end();
+        res.status(404).send({ message: 'User session not found' });
         return;
     }
-
-    if (!session_user){
-        res.status(401).send({message: 'Not authorized'});
+    if (!session_user) {
+        res.status(401).send({ message: 'Not authorized' });
         return;
     }
 
-    session_user.is_admin = isadmin;
-
-    let filter = {};
-    if(!session_user.is_admin) {
-        filter = {owner: session_user.uid};
+    let filter = { };
+    if(!is_admin) {
+        filter = { owner: session_user.uid };
     }
     let databases = await dbsrv.mongo_databases().find(filter).toArray();
     res.send(databases);
-    return;
-
 });
 
+
 router.get('/database/owner/:owner', async function(req, res) {
-    if(! req.locals.logInfo.is_logged) {
-        res.status(401).send({message: 'Not authorized'});
+    if(!req.locals.logInfo.is_logged) {
+        res.status(401).send({ message: 'Not authorized' });
         return;
     }
-    if(! sansrv.sanitizeAll([req.params.owner])) {
-        res.status(403).send({message: 'Invalid parameters'});
+    if(!sansrv.sanitizeAll([req.params.owner])) {
+        res.status(403).send({ message: 'Invalid parameters' });
         return;
     }
     let session_user = null;
-    let isadmin = false;
     try {
-        session_user = await dbsrv.mongo_users().findOne({_id: req.locals.logInfo.id});
-        isadmin = await rolsrv.is_admin(session_user);
+        session_user = await dbsrv.mongo_users().findOne({ _id: req.locals.logInfo.id });
     } catch(e) {
         logger.error(e);
-        res.status(404).send({message: 'User session not found'});
-        res.end();
+        res.status(404).send({ message: 'User session not found' });
         return;
     }
-
-    if (!session_user){
-        res.status(401).send({message: 'Not authorized'});
+    if (!session_user) {
+        res.status(401).send({ message: 'Not authorized' });
         return;
     }
 
```

Comment:
I think this route is unprotected: ie, I can list databases of another user.
Not really a security risk, but maybe add a check if session_id == owner || is_admin